vsintegration/update-vsintegration.cmd

```diff
@@ -3,30 +3,133 @@
+@rem   at which point everything will go through VSIXInstaller.exe
 
-if /i "%1" == "debug" goto :ok
-if /i "%1" == "release" goto :ok
+@echo off
+setlocal EnableDelayedExpansion
 
-echo Clobbers existing Visual Studio installation of F# bits
-echo Usage:
-echo    update-vsintegration.cmd debug 
-echo    update-vsintegration.cmd release 
-exit /b 1
+rem Count errors, warnings and succesful copies
+set ERRORCOUNT=0
+set WARNCOUNT=0
+set COPYCOUNT=0
+
+rem Enable colors, but can ONLY BE USED WITH PRINTING LINES THAT FIT IN A FILENAME!
+for /F "tokens=1,2 delims=#" %%a in ('"prompt #$H#$E# & echo on & for %%b in (1) do     rem"') do (
+  set "DEL=%%a"
+)
+
+if /i "%1" == "debug" (
+	set ACTION=debug
+	set DEPLOY=yes
+	set BINDIR=%~dp0..\%1\net40\bin
+	goto :ok
+)
+if /i "%1" == "release" (
+	set ACTION=release
+	set DEPLOY=yes
+	set BINDIR=%~dp0..\%1\net40\bin
+	goto :ok
+)
+if /i "%1" == "restore" (
+	set ACTION=restore
+	set DEPLOY=no
+	set BINDIR=%~dp0..\%1
+	goto :ok
+)
+if /i "%1" == "backup" (
+	set ACTION=backup
+	set DEPLOY=no
+	set BINDIR=%~dp0..\restore
+	goto :ok
+)
+
+set GOTOHELP=yes
 
 :ok
 
-set BINDIR=%~dp0..\%1\net40\bin
+set RESTOREDIR=%~dp0..\restore
 set TOPDIR=%~dp0..
 
+rem By using a token that does not exist in paths, this will resolve any ".." and "." in the path, even if path contains spaces
+FOR /F "tokens=*" %%I IN ("%RESTOREDIR%") DO set RESTOREDIR=%%~fI
+FOR /F "tokens=*" %%I IN ("%BINDIR%") DO set BINDIR=%%~fI
+FOR /F "tokens=*" %%I IN ("%TOPDIR%") DO set TOPDIR=%%~fI
+
+if /i "%GOTOHELP%" == "yes" goto :help
+GOTO :start
+
+
+:help
+
+echo.
+echo Clobbers existing Visual Studio installation of F# bits, like FSC compiler and FSI.
+echo After running, it will set the default FSC for use from Visual Studio 2017 to your
+echo locally built version. Requires administrator rights in most cases.
+echo.
+echo Syntax: %0 [debug^|release^|restore^|backup]
+echo.
+echo   debug     integrates debug builds of FSC, FSI ^& tools
+echo   release   integrates release builds of FSC, FSI ^& tools
+echo   restore   restores original SDK from an earlier backup
+echo   backup    backups the files that would be overwritten, does not deploy anything
+echo.
+echo Root location:        %TOPDIR%
+echo Debug bin location:   %TOPDIR%\debug\net40\bin
+echo Release bin location: %TOPDIR%\release\net40\bin
+echo Backup location:      %RESTOREDIR%
+echo.
+echo Clearing the git repository may clear the backup directory. To be on the safe
+echo side, you should place a copy of the backup dir outside of the git repo.
+echo.
+echo After installation you can check whether the installation succeeded by running the 
+echo following script in FSI in your Visual Studio 2017 installation, the output will
+echo show you the date and time the FSI executable was built.
+echo.
+echo ^> open System;;
+echo ^> open System.IO;;
+echo ^> open System.Reflection;;
+echo ^> printfn "%%A" (File.GetCreationTime(Assembly.GetEntryAssembly().Location));;
```

Comment:
Note to self: this should be fixed. It will show the creationtime of the FSI assembly that is shadow-copied, I believe, which has a different creation time. Furthermore, the FSI executable in VS is taken from the VSIX, not from the SDK, so this would yield to a false positive (unless you also executed the VSIX installer).

Reply:
Fixed in the latest push to this PR.